### PR TITLE
PES-1462 Allow button theme to toggle shadow on hover

### PR DESCRIPTION
**Pull request by ryannylander-medlypharmacy**
### PR Checklist

Please check if your PR fulfills the following requirements:

-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

### PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [X] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [ ] Performance improves
-   [ ] Adding missing tests
-   [X] Documentation content changes
-   [ ] Other... Please describe:

### What is the current behavior?

Solid and Outlined buttons have a box shadow that appears when the user hovers the mouse over them.

### What is the new behavior?

I am making the shadow on button hover optional in accordance with this ticket:
https://medlypharmacy.atlassian.net/browse/PES-1462
Shadow on hover will remain the default setting for buttons, and must be deactivated through the Medly-components theme.

### Does this PR introduce a breaking change?

-   [ ] Yes
-   [X] No

### Additional context

Screen Shot 2021-06-10 at 3 11 08 PM

### CODE REVIEW

File: packages/theme/src/core/button/types.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { FontVariants } from '../font/types';
 
 export type ButtonSizes = 'S' | 'M' | 'L';
 
-type Colors = {
+export type Colors = {
```
Comment (gmukul01) [inline packages/theme/src/core/button/types.ts:5]: We should not export this. This type is only used in this file.

File: packages/core/src/components/Button/styles/outlinedButton.tsx [hunk 2]
```diff
@@ -17,8 +17,18 @@ const getStyle = (textColor: string, bgColor = 'transparent', borderColor: strin
     }
 `;
 
+const getHoverShadow = (hoverShadow: boolean, shadowColor: string ) => {
```
Comment (gmukul01) [inline packages/core/src/components/Button/styles/outlinedButton.tsx:20]: You can write this like below
```ts
const getHoverShadow = (shadowColor: string ) => css`
      box-shadow: 0 0.2rem 0.8rem ${rgba(shadowColor, 0.35)};
    `
```

and then you can use is like 
```ts
${hoverShadow && getHoverShadow(borderColor?.hovered || textColor.hovered)}
```

File: packages/core/src/components/Button/styles/solidButton.tsx [hunk 3]
```diff
@@ -13,8 +13,18 @@ const getStyle = (textColor: string, bgColor: string) => css`
     }
 `;
 
+const getHoverShadow = (hoverShadow: boolean, bgColor: string) => {
```
Comment (gmukul01) [inline packages/core/src/components/Button/styles/solidButton.tsx:16]: You can refactor this like above

**Review by gmukul01 — CHANGES_REQUESTED** [review]

File: packages/core/src/components/Button/styles/outlinedButton.tsx (hunk 2 above)
Comment (ryannylander-medlypharmacy) [reply packages/core/src/components/Button/styles/outlinedButton.tsx]: Thanks, that's much cleaner, I have implemented it that way.

File: packages/theme/src/core/button/types.ts (hunk 1 above)
Comment (ryannylander-medlypharmacy) [reply packages/theme/src/core/button/types.ts]: Left this in by mistake, thanks, it's removed

**Review by jaredgiangrasso — APPROVED** [review]

**Review by gmukul01 — APPROVED** [review]
LGTM

**Pull request merged**